### PR TITLE
feat: allow users to customize alarm annotation properties

### PR DESCRIPTION
**Pull request by voho**
Allow users to override 1) color 2) label 3) visibility of the alarm annotation.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: lib/common/alarm/IAlarmAnnotationStrategy.ts [hunk 1]
```diff
@@ -36,9 +39,21 @@ export abstract class FillingAlarmAnnotationStrategy
...
+    if (props.overrideAnnotationLabel) {
+      annotation = { ...annotation, label: props.overrideAnnotationLabel };
+    }
+    if (props.overrideAnnotationVisibility) {
+      annotation = {
+        ...annotation,
+        visible: props.overrideAnnotationVisibility,
+      };
     }
     return annotation;
```
Comment (echeung-amzn) [inline lib/common/alarm/IAlarmAnnotationStrategy.ts:58]:
```
    return {
      ...this.createAnnotationToFill(props),
      ...(props.fillAlarmRange && { fill: this.getAlarmingRangeShade(props) }),
      ...(props.overrideAnnotationColor && { color: props.overrideAnnotationColor }),
      ...(props.overrideAnnotationLabel && { label: props.overrideAnnotationLabel }),
      ...(props.overrideAnnotationVisibility && { visible: props.overrideAnnotationVisibility }),
    };
```

Comment (voho) [reply lib/common/alarm/IAlarmAnnotationStrategy.ts:58]: Nice solution, TY.

**Review by echeung-amzn — COMMENTED** [review]
Can you add some tests to `AlarmFactory.test.ts` as well?

Otherwise, LGTM.

**Comment by voho** [comment]
> Can you add some tests to `AlarmFactory.test.ts` as well?
> 
> Otherwise, LGTM.

Sure, will do. Thanks.

**Review by echeung-amzn — APPROVED** [review]

**Pull request merged**